lib/utils/checkInvalidCLIOptions.js

```diff
@@ -0,0 +1,63 @@
+/* @flow */
+"use strict";
+
+const _ = require("lodash");
+const chalk = require("chalk");
+const EOL = require("os").EOL;
+const leven = require("leven");
+
+/*:: type allowedOptionsType = { [key: string]: { alias?: string } }*/
+
+const buildAllowedOptions = (allowedOptions /*: allowedOptionsType */) => {
+  let options = Object.keys(allowedOptions);
+  options = options.reduce((opts, opt) => {
+    const alias = allowedOptions[opt].alias;
+    if (alias) {
+      opts.push(alias);
+    }
+    return opts;
+  }, options);
+  options.sort();
+  return options;
+};
+
+const suggest = (all /*: string[]*/, invalid /*: string*/) => {
+  const sameLength = all.filter(option => option.length === invalid.length);
+  let suggestion = sameLength.find(option => leven(option, invalid) <= 1);
+  if (!suggestion) {
+    suggestion = sameLength.find(option => leven(option, invalid) <= 2);
+  }
+  if (!suggestion) {
+    suggestion = all.find(option => leven(option, invalid) <= 2);
+  }
+  if (!suggestion) {
+    suggestion = all.find(option => leven(option, invalid) <= 3);
+  }
+  return suggestion;
+};
```

Comment:
This `suggest` algorithm seems not good to me... 😓 

```
$ ./bin/stylelint.js --hel
Invalid option "--hel". Did you mean "-h"?
```

### Ideal

```
$ ./bin/stylelint.js --hel
Invalid option "--hel". Did you mean "--help"?
```

Reply:
Yep, the latter is better.